language-bindings/rust/src/instance.rs

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2019 Intel Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+//! an instantiated module. The module is instantiated with the given imports.
+//! get one via `Instance::new()`
+
+#![allow(unused_variables)]
+
+use ::core::ffi::c_char;
+
+use wamr_sys::{
+    wasm_module_inst_t, wasm_runtime_deinstantiate, wasm_runtime_init_thread_env,
+    wasm_runtime_instantiate,
+};
+
+use crate::{
+    helper::error_buf_to_string, helper::DEFAULT_ERROR_BUF_SIZE, module::Module, RuntimeError,
+};
+
+#[derive(Debug)]
+pub struct Instance {
+    instance: wasm_module_inst_t,
+}
+
+impl Instance {
+    /// instantiate a module with stack size
+    ///
+    /// # Error
+    ///
+    /// Return `RuntimeError::CompilationError` if failed.
+    pub fn new(module: &Module, stack_size: u32) -> Result<Self, RuntimeError> {
+        Self::new_with_args(module, stack_size, 0)
+    }
+
+    /// instantiate a module with stack size and host managed heap size
+    ///
+    /// heap_size is used for `-nostdlib` Wasm and wasm32-unknown
+    ///
+    /// # Error
+    ///
+    /// Return `RuntimeError::CompilationError` if failed.
+    pub fn new_with_args(
+        module: &Module,
+        stack_size: u32,
+        heap_size: u32,
+    ) -> Result<Self, RuntimeError> {
+        let init_thd_env = unsafe { wasm_runtime_init_thread_env() };
```

Comment:
Should call `wasm_runtime_destroy_thread_env` somewhere?

Reply:
nice catch. it is a good one,